devito/finite_differences/coefficients.py

```diff
@@ -87,8 +91,6 @@ def weights(self):
     def _check_input(self, deriv_order, function, dimension, weights):
         if not isinstance(deriv_order, int):
             raise TypeError("Derivative order must be an integer")
-        # NOTE: Can potentially be tidied up following the implementation
-        # of lazy evaluation.
```

Comment:
why removed?

Reply:
It didn't seem immediately obvious to me that the lazy implementation could help in tidying this up. I'm very possibly missing something though?